### ISSUE TITLE
Allow configuring FIND_LINKS with an environment variable

### ISSUE THREAD

**Issue by alexandre-fabra** [astral-sh/uv#1839]
I would like to be able to configure the `find-links` URL with an environment variable similar to how `PIP_FIND_LINKS` works.

To verify the current behaviour, I used the following snippet:

```bash
export UV_INDEX_URL=https://index.url \
&& export UV_EXTRA_INDEX_URL=https://extra.index.url \
&& export UV_FIND_LINKS=https://find.links \
&& uv pip install --help
```

and then verified that the env values for `INDEX_URL` and `EXTRA_INDEX_URL` are honored, but not for `FIND_LINKS`:

```
  -i, --index-url <INDEX_URL>
          The URL of the Python package index (by default: https://pypi.org/simple)
          
          [env: UV_INDEX_URL=https://index.url]

      --extra-index-url <EXTRA_INDEX_URL>
          Extra URLs of package indexes to use, in addition to `--index-url`
          
          [env: UV_EXTRA_INDEX_URL=https://extra.index.url]

  -f, --find-links <FIND_LINKS>
          Locations to search for candidate distributions, beyond those found in the indexes.
          
          If a path, the target must be a directory that contains package as wheel files (`.whl`) or source distributions (`.tar.gz` or `.zip`) at the top level.
          
          If a URL, the page must contain a flat list of links to package files.

```

**Comment by hauntsaninja** [comment]
This is maybe duplicated by the more general https://github.com/astral-sh/uv/issues/1789

**Comment by charliermarsh** [comment]
Agree, want to add these.

**Comment by garthk** [comment]
I regret #1789 was closed without a UV_FIND_LINKS:

> Closing for now as we have many of these.
> ―  @charliermarsh in https://github.com/astral-sh/uv/issues/1789#issuecomment-2066674786

**Comment by zanieb** [comment]
Thanks for noting.

**Comment by konstin** [comment]
This was fixed in #7912, adding support for `UV_FIND_LINKS`.
